### PR TITLE
docs: Implementing authentication in graphql

### PR DESCRIPTION
**Pull request by shegz101**
Here is an outline for the topic" implementing authentication in graphql." I systematically created an outline that will aid the writer in quickly writing a good article on this topic. The outline contains all the major concepts and practical implementation of GraphQl authentication.

### CODE REVIEW

**Comment by shegz101** [comment]
@Frenzyritz13 Amit said I should reach out to you and that you are the one in charge of reviewing PRs from now on. Thank you!

**Comment by shegz101** [comment]
@tusharmath please you can try re-merging it now. All checks have passed. Thank you! Just so you know, this is the second PR to create an outline for the topic above, which is one of the topics @amitksingh1490 gave to me.

**Comment by shegz101** [comment]
> @tusharmath please you can try re-merging it now. All checks have passed. Thank you! Just so you know, this is the second PR to create an outline for the topic above, which is one of the topics @amitksingh1490 gave to me.

I think @Frenzyritz13 is in charge of reviewing it. Thank you!

File: outlines/implementing-authentication-in-graphql.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+- Mutation types (signup, login, logout)
+
+- Query types
+
+- Custom types (User, AuthPayload)
+
+C. Creating/Implementing Resolvers
+
+D. Server-Side Implementation (Apollo Server and JWT)
```
Comment (amitksingh1490) [inline outlines/implementing-authentication-in-graphql.md]:
```suggestion
D. Server-Side Implementation (Tailcall Server and JWT)
```

File: outlines/implementing-authentication-in-graphql.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+A. Defining Authenticated Fields and Types
+
+B. Using Directives for Authorization
+
+## Authentication Workflow
+
+A. Client-Side Authentication
+
+```grapql
```
Comment (amitksingh1490) [inline outlines/implementing-authentication-in-graphql.md]:
```suggestion
```graphql
```

File: outlines/implementing-authentication-in-graphql.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+# code snippet to explain Client side Authntication
+```
+
+B. Server-Side Authentication
+
+```grapql
```
Comment (amitksingh1490) [inline outlines/implementing-authentication-in-graphql.md]:
```suggestion
```graphql
```

File: outlines/implementing-authentication-in-graphql.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+## Conclusion
+
+A. Summary of Key Points
+
+## Resources
+
+A. Apollo Official Documentation
+
+B. GraphQL Official Documentation
```
Comment (amitksingh1490) [inline outlines/implementing-authentication-in-graphql.md]:
```suggestion

B. GraphQL Official Documentation
```

File: outlines/implementing-authentication-in-graphql.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+A. Authentication Packages
+
+- Apollo server packages
+
+- GraphQL packages
+
+B. Monitoring and Observability
+
+- Tools used for tracing, logging, and performance monitoring
```
Comment (amitksingh1490) [inline outlines/implementing-authentication-in-graphql.md]: use tailcall

**Review by amitksingh1490 — CHANGES_REQUESTED** [review]

**Comment by amitksingh1490** [comment]
Added comments, moving it to draft

**Comment by shegz101** [comment]
@amitksingh1490 I have implemented the changes.

**Comment by shegz101** [comment]
> Added comments, moving it to draft

Thank you. I have committed the changes.

**Comment by amitksingh1490** [comment]
This PR has unnecessary file changes. Please keep the changes limited to the scope of work of this PR

**Comment by shegz101** [comment]
> This PR has unnecessary file changes. Please keep the changes limited to the scope of work of this PR

Yes, I am having some issues with that. Seems files from other branches are being merged. I will delete/close this PR, then create a new one and tag you.

**Comment by shegz101** [comment]
> This PR has unnecessary file changes. Please keep the changes limited to the scope of work of this PR

@amitksingh1490 I have removed the unnecessary file changes. Thank you!

File: outlines/implementing-authentication-in-graphql.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+B. Verifying Tokens or Sessions
+
+C. Populating Request Context
+
+### Authentication Resolvers
+
+A. Accessing Authenticated User Information
+
+B. Implementing Authorization Logic
```
Comment (amitksingh1490) [inline outlines/implementing-authentication-in-graphql.md:41]:
```suggestion
B. Using Tailcall Authentication directive
```

**Pull request closed without merging**